### PR TITLE
Added  PluginBase::getResourceContents

### PR DESCRIPTION
**Pull request by alejandroliu**
Adds a convenient function to PluginBase called getResourceContents.

It is the same idea as file_get_contents, but for resources.

### CODE REVIEW

**Comment by Falkirks** [comment]
:+1:

**Comment by iksaku** [comment]
Good idea! :+1:

**Comment by 64FF00** [comment]
:+1:

File: src/pocketmine/plugin/PluginBase.php [hunk 1]
```diff
@@ -187,6 +187,23 @@ public function getResource($filename){
 	}
 
 	/**
+	 * Gets the contents of an embedded resource on the plugin file.
+	 *
+	 * @param string $filename
+	 *
+	 * @return string, or null
```
Comment (sekjun9878) [inline src/pocketmine/plugin/PluginBase.php]: This is not a valid PHPDoc comment.

**Comment by alejandroliu** [comment]
Sorry. Don't really now what's needed to make a valid PHPdoc.

File: src/pocketmine/plugin/PluginBase.php (hunk 1 above)
Comment (PEMapModder) [inline src/pocketmine/plugin/PluginBase.php]: `string|null`
Refer to other parts of PocketMine code.

**Comment by alejandroliu** [comment]
Sooo... took the wrong example from Line 78 (https://github.com/alejandroliu/PocketMine-MP/blob/PluginBase-suggestion/src/pocketmine/plugin/PluginBase.php#L178)

**Comment by shoghicp** [comment]
Basically, this is a shortcut :)

In first place, this was intended to be a [SplFileObject](https://php.net/manual/en/class.splfileobject.php), but it lacks something very basic: ability to get the full contents. It also contains a lot of... interesting methods, that would fail or crash on phar archives.

**Comment by alejandroliu** [comment]
So does that mean that you won't be merging this?

**Comment by PEMapModder** [comment]
@shoghicp Then why not `PharFileInfo`? Its functions are quite interesting as well.

**Comment by shoghicp** [comment]
@PEMapModder because that is dependent on the plugin being a phar, and that is only true in 1/3 of the possible plugin formats right now :)

**Comment by alejandroliu** [comment]
@shoghicp, So in the meantime, can you merge this?

The other option to the @PEMapModder suggestion would be that the PluginLoader provides a factory to create the appropriate xxxFileInfo object.  So the PharLoader would return PharFileObject, the Folder loader would return SplFileInfo.  Any other loaders can return whatever is necessary.  The PHP snipped loader shouldn't throw and error as it wouldn't make sense for it.

**Comment by shoghicp** [comment]
@alejandroliu PharFileInfo is just an SplFileInfo with the same issues as the original object - you can't get the full data directly. That's why that was not chosen.

**Comment by PEMapModder** [comment]
Write your own class :stuck_out_tongue_closed_eyes:

**Comment by alejandroliu** [comment]
I think just implementing the proposing getResourceContents from this PullRequest would take care of most use cases by most plugin developers.

**Comment by sekjun9878** [comment]
I agree that a simple solution is more concise and accessible than trying to tie this into an object.

**Comment by sekjun9878** [comment]
Doesn't look like this will cause any problems and it's BC compatible, so I'm merging this.

**Comment by shoghicp** [comment]
This was not decided to be merged. I've force-pushed master back.

**Comment by sekjun9878** [comment]
Why?

**Comment by iksaku** [comment]
D:

**Pull request merged**